extide/gradle/src/org/netbeans/modules/gradle/GradleJavaCompatProblemsProvider.java

```diff
@@ -143,8 +143,12 @@ private static int getJavaMajorVersion(File javaHome) {
 
             }
             String javaVersion = releasePros.getProperty("JAVA_VERSION"); //NOI18N
+            // This should look like "17" or "17.0.9"
             if ((javaVersion != null) && javaVersion.startsWith("\"") && javaVersion.endsWith("\"")) {
-                javaVersion = javaVersion.substring(1, javaVersion.indexOf('.'));
+                int dot = javaVersion.indexOf('.');
+                javaVersion = dot > 0
+                        ? javaVersion.substring(1, javaVersion.indexOf('.'))
+                        : javaVersion.substring(1, javaVersion.length() - 1);
```

Comment:
this could be using:
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Runtime.Version.html#parse(java.lang.String)

once the module moved to 11 so that we don't have to come up with so many ways how to parse java versions.
Maybe mark the spot with a TODO?